doc/docs/master/enterprise/auth/oidc/configure-keycloak.md

```diff
@@ -0,0 +1,160 @@
+  ```
+
+  **System Response:**
+
+  ```bash
+  Unable to find image 'quay.io/keycloak/keycloak:10.0.1' locally
+  10.0.1: Pulling from keycloak/keycloak
+  e96e3a1df3b2: Pull complete
+  1b99828eddf5: Pull complete
+  ...
+
+  19:08:49,501 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 63) WFLYUT0021: Registered web context: '/auth' for server 'default-server'
+  19:08:49,793 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 46) WFLYSRV0010: Deployed "keycloak-server.war" (runtime-name : "keycloak-server.war")
+  ```
+
+  In the above command, we use Keycloak version 10.0.1. You might need to
+  update this version to the latest Keycloak build. The user and password
+  are both set to `admin`. Booting Keycloak might take some time.
+
+  If you are deploying in production, see [Keycloak Documentation](https://www.keycloak.org/getting-started).
+
+## Configure Keycloak
+
+After you have installed Pachyderm and Keycloak, follow the steps
+below to configure the two components to work together.
+
+!!! note
+    If you have used a different authentication type before,
+    you need to disable authentication before configuring
+    Keycloak. Run `pachctl auth deactivate`.
+
+To configure Pachyderm with Keycloak, complete the following steps:
+
+1. Go to the **Keycloak Administration Console** by pointing your browser to
+`http://localhost:8080/auth/`and logging in with your user and password.
+If you have boot Keycloak in Docker as described above, use `admin` as
+login and password.
+
+   ![Keycloak login](../../../assets/images/s_keycloak_login.png)
+
+1. If you do not have one already, add a new realm. Otherwise, proceed to
+the next step.
+
+   A realm is a logically isolated group, similar to a tenant. By default,
+   Keycloak predefines a `Master` realm for administrative purposes. You need
+   to create a new realm for Pachyderm.
+
+1. If you do not have any users to whom you want to grant access to Pachyderm,
+create a user by clicking **Users > Add user** and filling out the required
+fields. Otherwise, proceed to the next step.
+
+   1. Set up a password for this user by clicking **Credentials** and typing
+   your password information.
+
+      You might want to set the temporary password toggle to off so that your
+      password does not expire.
+
+1. Create a client for Pachyderm:
+
+   1. Click **Clients > Create**.
+   1. Specify a client ID. For example, *Pachyderm*.
+   1. Set the **Client Protocol** to **openid-connect**.
+   1. Set the **Root URL** to `http://<ip>:14687/authorization-code/callback`.
+      For example:
+
+      ```bash
+      http://localhost:14687/authorization-code/callback
+      ```
+
+      You can configure additional settings, such as require consent and so on,
+      as needed, but this minimal configuration is everything you need to make
+      Keycloak work with Pachyderm.
+
+      ![Keycloak Client](../../../assets/images/s_keycloak_client.png)
+
```

Comment:
If you're running Keycloak in Docker for Mac, there's a bug (https://github.com/docker/for-mac/issues/2670) which will force your browser and Pachyderm to talk to keycloak with different URLs (i.e. localhost vs 172.17.0.2). Keycloak will notice this discrepancy, and will fail the token validation. There are some workarounds described in the issue I linked, but I haven't tested them myself yet.

Reply:
I've changed the instructions to use minikube instead and there are no issues.

Reply:
Should we still note this bug in docker for Mac?

Reply:
I'd rather avoid adding this in the docs as this is out of our control and can be fixed any minute. I'd avoid adding links to bugs in the docs as they will quickly become obsolete.